Source/Initialization/WarpXInitData.cpp

```diff
@@ -266,6 +267,74 @@ WarpX::ComputePMLFactors ()
     }
 }
 
+void
+WarpX::ComputeMaxStep ()
+{
+    if (do_compute_max_step_from_zmax) {
+        computeMaxStepBoostAccelerator(geom[0]);
+    }
+
+    // Make max_step and stop_time self-consistent, assuming constant dt.
+
+    // If max_step is the limiting condition, decrease stop_time consistently
+    if (stop_time > t_new[0] + dt[0]*(max_step - istep[0]) ) {
+        stop_time = t_new[0] + dt[0]*(max_step - istep[0]);
```

Comment:
We call this only once in `Init`, right?

Because I was wondering with that logic, if we would update `stop_time` and `max_step` multiple times due to rounding errors. Otherwise seems legit :)